### PR TITLE
Update the shift mechanism to keep the stage is within the ratio

### PR DESCRIPTION
**Pull request by houshengbo**
# Changes

- Each time the adjustment mechanism starts to work, the largest traffic delta should be the over-consumption-ratio.
- When the old revision scale down to take 0% of the traffic, the SPA sets the min and max scales to 0.
- Per the revisions scaling up, we pick up the greater value as the target replicas' number between the results adding a delta and calculating by the replicas/traffic ratio. Larger number means to set a higher standard for the revision to scale up.
- RolloutOrchestrator should be created after the configuration.

/kind <kind>

Fixes #

**Release Note**

```release-note

```

**Docs**

```docs

```

### CODE REVIEW

File: pkg/reconciler/rolloutorchestrator/rolloutorchestrator.go [hunk 1]
```diff
@@ -147,6 +148,8 @@ func (r *Reconciler) ReconcileKind(ctx context.Context, ro *v1.RolloutOrchestrat
 				}
 			}
 		} else if len(ro.Spec.InitialRevisions) != 0 {
+			fmt.Println("scale down none, but have InitialRevisions")
```
Comment (yuzisun) [inline pkg/reconciler/rolloutorchestrator/rolloutorchestrator.go:151]: I guess these are debugging statements which can be removed

File: pkg/reconciler/rolloutorchestrator/rolloutorchestrator.go [hunk 2]
```diff
@@ -147,6 +148,8 @@ func (r *Reconciler) ReconcileKind(ctx context.Context, ro *v1.RolloutOrchestrat
...
+			fmt.Println(ro.Spec.InitialRevisions[0].RevisionName)
```
Comment (yuzisun) [inline pkg/reconciler/rolloutorchestrator/rolloutorchestrator.go:152]: Same here

File: pkg/reconciler/rolloutorchestrator/rolloutorchestrator.go [hunk 3]
```diff
@@ -254,32 +259,38 @@ func UpdateSPAForRevDown(spa *v1.StagePodAutoscaler, revision *v1.TargetRevision
 	if !scaleUpReady {
 		return spa
 	}
+
+	if len(strings.TrimSpace(revision.RevisionName)) == 0 {
```
Comment (yuzisun) [inline pkg/reconciler/rolloutorchestrator/rolloutorchestrator.go:263]: Revision name can be empty ?

Comment (houshengbo) [reply pkg/reconciler/rolloutorchestrator/rolloutorchestrator.go:263]: It is not very possible to happen in real time, but I have a test case on it for safeguard reasons.

**Pull request merged**